Assets/MRTK/SDK/Editor/Inspectors/UX/InteractiveElement/TrackedStatesInspector.cs

```diff
@@ -0,0 +1,282 @@
+        private void DrawStateEventsScriptableSubEditor(SerializedProperty scriptable)
+        {
+            if (scriptable.objectReferenceValue != null)
+            {
+                UnityEditor.Editor configEditor = UnityEditor.Editor.CreateEditor(scriptable.objectReferenceValue);
+                
+                using (new EditorGUILayout.VerticalScope(InspectorUIUtility.Box(10)))
+                {
+                    EditorGUILayout.Space();
+                    configEditor.OnInspectorGUI();
+                    EditorGUILayout.Space();
+                }
+            }
+        }
+
+        private void SetCoreStateType(SerializedProperty stateProp, SerializedProperty stateNameProp)
+        {
+            Rect position = EditorGUILayout.GetControlRect();
+            using (new EditorGUI.PropertyScope(position, new GUIContent("State"), stateProp))
+            {
+                string[] coreInteractionStateNames = Enum.GetNames(typeof(CoreInteractionState)).ToArray();
+                int id = Array.IndexOf(coreInteractionStateNames, -1);
+                int newId = EditorGUI.Popup(position, id, coreInteractionStateNames);
+
+                // NOTE FOR THE FUTURE: Sort the core states in a menu that indicates whether a core state is a near 
+                // interaction state, far interaction state, or both to futher push the mental model of the MRTK interaction model
+
+                if (newId != -1)
+                {
+                    string selectedState = coreInteractionStateNames[newId];
+
+                    // If this state is not already being tracked then change the name
+                    if (!instance.IsStateTracked(selectedState))
+                    {
+                        stateNameProp.stringValue = selectedState;
+                    }
+                    else
+                    {
+                        // If a state that is already being tracked is selected, reset the id until a 
+                        // state that is not being tracked is selected
+                        newId = -1;
+                    }
+                }
+            }
+        }
+
+        // Add a new state to track.  
+        private SerializedProperty AddNewState(string initialStateName)
+        {
+            stateList.InsertArrayElementAtIndex(stateList.arraySize);
+
+            SerializedProperty newState = stateList.GetArrayElementAtIndex(stateList.arraySize - 1);
+            SerializedProperty name = newState.FindPropertyRelative("stateName");
+            SerializedProperty eventConfiguration = newState.FindPropertyRelative("eventConfiguration");
+
+            eventConfiguration.objectReferenceValue = null;
+
+            // Set the initial name of the state added.  This name will be changed once a core state is selected from the list or after a new 
+            // name is set via text field.
+            name.stringValue = initialStateName;
```

Comment:
Since you are using strings to encode state names already, I don't see much of a reason to need the backing data type to be a enum with a fixed number of values. You could instead allow end-users to configure the states they want for interactive elements.

If you instead want to use an enum, then I don't see much of a reason to have the backing data type be a string -- I would instead use a int (less string comparisons, and also allows you to change names in the future).